src/Run/Setup.cs

```diff
@@ -14,13 +18,51 @@ public class Setup
         public string Os { get; set; }
         public string ConfigurationFilePath { get; set; }
 
+        private string ParseSettingValue(string inputValue)
+        {
+            string value = string.Empty;
+            int length = inputValue.Length;
+            for (int i = 0; i < length; i++)
+            {
+                if (inputValue[i] == '$')
```

Comment:
So the idea of the string format is it can be anything like, ```${OSVersion}-${CPUArch}``` or ```myOS.${OSName}```.. This code will just replace all occurrences of ${} in the string with the value from SettingValueProvider if available.